app/controllers/concerns/questionnaire_renderable.rb

```diff
@@ -1,12 +1,11 @@
 module QuestionnaireRenderable
   extend ActiveSupport::Concern
 
-  def prep_questionnaire
+  def prep_questionnaire(json: true)
     @attrib = params[:filter] || "criteria"
     @response_set ||= @loan.send(@attrib) || LoanResponseSet.new(kind: @attrib, loan: @loan)
-    @roots = @response_set.loan_question_set.root_group_preloaded
-    @questions_json = @roots.children_applicable_to(@loan).map do |i|
-      LoanQuestionSerializer.new(i, loan: @loan)
-    end.to_json
+    root = @response_set.loan_question_set.root_group_preloaded
+    @top_level_questions = LoanFilteredQuestion.new(root, loan: @loan).children
+    @questions_json = @top_level_questions.map { |q| LoanQuestionSerializer.new(q) }.to_json if json
```

Comment:
Some day hopefully we can move away from doing the top level questions thing and just load the root and kick things off with that. This is a definite step in the right direction!!

Reply:
Done. 😛